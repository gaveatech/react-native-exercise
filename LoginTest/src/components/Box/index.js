import React from 'react';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';

const Box = ({size, bgColor, align, children}) => {
  return <Wrapper {...{size, bgColor, align}}>{children}</Wrapper>;
};

const Wrapper = styled.View`
  background-color: ${props => props.bgColor};
  padding: 15px;
  width: ${props => props.size};
  border-radius: 15px;
  shadow-color: #000;
  shadow-offset: {width: 0, height: 2};
  shadow-opacity: 0.8;
  shadow-radius: 2;
  elevation: 1;
  align-items: ${props =>
    props.align === 'center'
      ? 'center'
      : props.align === 'left'
      ? 'flex-start'
      : 'flex-end'};
`;

Box.propTypes = {
  size: PropTypes.string,
  bgColor: PropTypes.string,
  align: PropTypes.string,
  children: PropTypes.node.isRequired,
};

Box.defaultProps = {
  size: '100%',
  align: 'center',
  bgColor: '#fff',
};

export default Box;
