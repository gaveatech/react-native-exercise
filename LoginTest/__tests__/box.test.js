import React from 'react';
import {Text} from 'react-native';
import renderer from 'react-test-renderer';
import Box from '../src/components/Box';

test('<Box> renders', () => {
  const tree = renderer
    .create(
      <Box>
        <Text>SÃ³ um teste</Text>
      </Box>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
