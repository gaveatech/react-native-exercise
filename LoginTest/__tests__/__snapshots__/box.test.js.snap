// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box> renders 1`] = `
<View
  align="center"
  bgColor="#fff"
  size="100%"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 15,
        "elevation": 1,
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 15,
        "shadowColor": "#000",
        "shadowOpacity": 0.8,
        "shadowRadius": 2,
        "width": "100%",
      },
    ]
  }
>
  <Text>
    SÃ³ um teste
  </Text>
</View>
`;
